### ISSUE TITLE
Assymetry in reducing log P(y | x, w) for `s_grads` and `w_grads`

### ISSUE THREAD

**Issue by shwang** [pomonam/NoisyNaturalGradient#2]
Hello,

I noticed that in [`regression/model.py`](https://github.com/pomonam/NoisyNaturalGradient/blob/master/regression/model.py#L119), `log_p_y_xw` which has shape `(n_particles, n_batch)` is reduced differently for calculating `s_grads` and `w_grads`.

`s_grads` reduces `log_p_y_xw` by summing over the `n_batch` axis:
```
s_grads = tf.gradients(tf.reduce_sum(sampled_log_prob), s)
```

but `w_grads` reduces `log_p_y_xw` by averaging over the `n_batch` axis:
```
w_grads = tf.gradients(tf.reduce_sum(tf.reduce_mean(log_py_xw, 1), 0), qws)
```

This asymmetry feels odd to me, is this the intended behavior?

**Comment by shwang** [comment]
Mainly interested in this question because I'd like to define a custom `model.log_py_xw` that is already reduced. In other words, I want to reduce `log_py_xw` ahead of time so that it is already a scalar by the time that `s_grads` and `w_grads` are calculated.

**Comment by shwang** [comment]
(For the record, using `reduce_sum` on both gradients resulted in NaNs)
